projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultConfiguration.java

```diff
@@ -5,27 +5,79 @@
 
 package com.vmware.taurus.secrets.service.vault;
 
+import com.vmware.taurus.exception.SecretStorageNotConfiguredException;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.vault.authentication.TokenAuthentication;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+import org.springframework.lang.Nullable;
+import org.springframework.vault.authentication.*;
 import org.springframework.vault.client.VaultEndpoint;
+import org.springframework.vault.config.AbstractVaultConfiguration;
 import org.springframework.vault.core.VaultOperations;
 import org.springframework.vault.core.VaultTemplate;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.DefaultUriBuilderFactory;
 
 import java.net.URI;
 import java.net.URISyntaxException;
 
 @Configuration
-public class VaultConfiguration {
+public class VaultConfiguration extends AbstractVaultConfiguration {
+
+  @Value("${vdk.vault.uri:}")
+  String vaultUri;
+
+  @Value("${vdk.vault.approle.roleid:}")
+  String roleId;
+
+  @Value("${vdk.vault.approle.secretid:}")
+  String secretId;
+
+  @Value("${vdk.vault.token:}")
+  @Nullable
+  String vaultToken;
 
   @Bean
   public VaultOperations vaultOperations(
-      @Value("${vdk.vault.uri:}") String vaultUri, @Value("${vdk.vault.token:}") String vaultToken)
-      throws URISyntaxException {
-    VaultEndpoint vaultEndpoint = VaultEndpoint.from(new URI(vaultUri));
-    TokenAuthentication clientAuthentication = new TokenAuthentication(vaultToken);
+      VaultEndpoint vaultEndpoint, SessionManager sessionManager) {
+
+    SimpleClientHttpRequestFactory clientHttpRequestFactory = new SimpleClientHttpRequestFactory();
+    return new VaultTemplate(vaultEndpoint, clientHttpRequestFactory, sessionManager);
+  }
+
+  @NotNull
+  @Override
+  @Bean
+  public VaultEndpoint vaultEndpoint() {
+    try {
+      return VaultEndpoint.from(new URI(this.vaultUri));
+    } catch (URISyntaxException e) {
+      throw new SecretStorageNotConfiguredException();
+    }
+  }
+
+  @NotNull
+  @Override
+  @Bean
+  public ClientAuthentication clientAuthentication() {
+    // Token authentication should only be used for development purposes. If the token expires, the
+    // secrets
+    // functionality will stop working
+    if (StringUtils.isNotBlank(vaultToken)) {
+      return new TokenAuthentication(vaultToken);
```

Comment:
I would add a warning that vault Token is being used . Reducing the chance of someone mistakenly using it in production. Or if someone uses that has issues , it's a hint toward the issue.

I am assuming this is called only once and the warning will be printed once.

Reply:
done. 
Assumption is correct.